app/jobs/course/assessment/answer/auto_grading_job.rb

```diff
@@ -6,12 +6,15 @@ class Course::Assessment::Answer::AutoGradingJob < ApplicationJob
 
   # Performs the auto grading.
   #
+  # @param [String|nil] redirect_to The path to be redirected after auto grading job was finished.
   # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store the grading
   #   results into.
-  def perform_tracked(auto_grading)
+  def perform_tracked(auto_grading, redirect_to = nil)
     instance = Course.unscoped { auto_grading.answer.question.assessment.course.instance }
     ActsAsTenant.with_tenant(instance) do
       Course::Assessment::Answer::AutoGradingService.grade(auto_grading)
     end
+
+    redirect_to redirect_to
```

Comment:
It works and I have specs, I think only the second redirect_to refers to the variable.